.docker/Dockerfile-CPU

```diff
@@ -4,18 +4,75 @@
 # flashlight       master     (git, CPU backend)
 # ==================================================================
 
-FROM flml/flashlight:cpu-base-consolidation-latest
-
-ENV MKLROOT="/opt/intel/mkl"
+FROM flml/flashlight:cpu-base-consolidation-latest as build
 
 # ==================================================================
 # flashlight with CPU backend
 # ------------------------------------------------------------------
 # Setup and build flashlight
-RUN mkdir /root/flashlight
+RUN mkdir /tmp/flashlight
+
+COPY . /tmp/flashlight
+
+RUN mkdir -p /tmp/flashlight/build && \
+    cd /tmp/flashlight/build && \ 
+    cmake -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX="/opt/flashlight" -DFL_BACKEND="CPU" -DFL_LIBRARIES_USE_CUDA="OFF" -DFL_BUILD_TESTS="ON" -DDNNL_DIR="/opt/onednn/lib/cmake/dnnl" -DGloo_INCLUDE_DIR="/opt/gloo/include" -DGloo_NATIVE_LIBRARY="/opt/gloo/lib/libgloo.a" .. && \
+    make -j$(nproc) && \
+    make install -j$(nproc)
+
+
 
-COPY . /root/flashlight
+#############################################################################
+#                            SECOND STAGE                                   #
+#############################################################################
 
-RUN cd /root/flashlight && mkdir -p build && \
-    cd build && cmake .. -DFL_BACKEND=CPU -DFL_LIBRARIES_USE_CUDA=OFF && \
-    make -j$(nproc) && make install
+FROM ubuntu:18.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# install dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        g++ \
+        # for cmake
+        zlib1g-dev libcurl4-openssl-dev \
+        # for MKL
+        apt-transport-https \
+        gpg-agent gnupg2 \
+        # for arrayfire CPU backend
+        # OpenBLAS
+        libopenblas-dev libfftw3-dev liblapacke-dev \
+        # ATLAS
+        libatlas3-base libatlas-base-dev libfftw3-dev liblapacke-dev \
+        # ssh for OpenMPI
+        openssh-server openssh-client \
+        # OpenMPI
+        libopenmpi-dev libomp-dev openmpi-bin \
```

Comment:
`libomp-dev` ≠ OpenMPI - it's the LLVM OpenMP runtime - and it could interact poorly with MKL's iomp which already fulfills that requirement

Reply:
will be solving in https://github.com/facebookresearch/flashlight/pull/401